Class 2.ipynb

```diff
@@ -9,45 +9,252 @@
+    "#### Notebook Basics\n",
+    "\n",
+    "Notebooks are a very convenient way for organizing and, most importantly, sharing data analyses. They allow for single-purpose explanatory scripts to be shared (complete with visualizations). They are often used in research settings or for prototyping. The ability to intersperse code with formatted, explanatory markdown is especially useful.\n",
     "\n",
-    "interactive data analysis tricks"
+    "Basic shortcuts and UI:\n",
+    "* Edit a cell by hitting \"enter\"\n",
+    "* Switch out of edit  mode by hitting \"esc\"\n",
+    "* Run a cell with cmd+\"enter\"\n",
+    "* Add a new cell above with \"a\" (when not in edit mode)\n",
+    "* Add a new cell below with \"b\" (when not in edit mode)\n",
+    "\n",
+    "You can find more information [in the docs](http://jupyter-notebook.readthedocs.io/en/latest/examples/Notebook/Notebook%20Basics.html)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Load 311 data\n",
+    "```bash\n",
+    "$ cd ~/workspace # or wherever you store working repos\n",
+    "$ git clone https://github.com/jvns/pandas-cookbook\n",
+    "$ cd python-for-data-analysis\n",
+    "```\n",
+    "    "
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 49,
+   "execution_count": 4,
    "metadata": {
     "collapsed": true
    },
    "outputs": [],
    "source": [
-    "import pandas as pd\n",
-    "import numpy as np\n",
+    "import pandas as pd # use 'as' keyword to namespace a package\n",
+    "complaints = pd.read_csv('../pandas-cookbook/data/311-service-requests.csv', low_memory=False)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(complaints.head())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(len(complaints))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(complaints.columns)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(complaints['Created Date'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "print(complaints[0:3])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "print(complaints['Location'].dtype)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Exercise\n",
+    "Write a loop that will loop through the columns of the data frame printing their names and types"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Plotting Basics with groupby"
```

Comment:
I think it might make sense to move this to after you've explained what indices and group by are.